nodeadm/internal/kubelet/eni_max_pods.go

```diff
@@ -1,67 +1,119 @@
 package kubelet
 
 import (
+	"bufio"
+	"bytes"
 	"context"
 	_ "embed"
-	"strconv"
-	"strings"
+	"encoding/json"
+	"fmt"
+	"math"
 
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/awslabs/amazon-eks-ami/nodeadm/internal/util"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
 	"go.uber.org/zap"
 )
 
+const (
+	defaultENIsVar = "default_enis"
+	ipsPerENIVar   = "ips_per_eni"
+	maxPodsVar     = "max_pods"
+)
+
 // default value from kubelet
 // https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration
 const defaultMaxPods = 110
 
-//go:embed eni-max-pods.txt
-var eniMaxPods string
-
-var MaxPodsPerInstanceType map[string]int
+//go:embed instance-info.jsonl
+var cachedInstanceInfoBytes []byte
 
-func init() {
-	MaxPodsPerInstanceType = make(map[string]int)
-	lines := strings.Split(eniMaxPods, "\n")
-
-	for _, line := range lines {
-		line = strings.TrimSpace(line)
-		if line == "" || strings.HasPrefix(line, "#") {
+func GetInstanceInfo(ctx context.Context, awsRegion string, instanceType string) (util.InstanceInfo, error) {
+	// try to read it from the cached file first
+	cachedInfoReader := bytes.NewReader(cachedInstanceInfoBytes)
+	s := bufio.NewScanner(cachedInfoReader)
+	for s.Scan() {
```

Comment:
```suggestion
	for s := bufio.NewScanner(bytes.NewReader(cachedInstanceInfoBytes)); s.Scan(); {
```

i typically try hold back purely styling nits, but i enforce this pattern to avoid polluting variable scopes